### DISCUSSION TITLE
How to integrate @svgr/webpack to make turbopack work?

### DISCUSSION THREAD

**Discussion by penteleichuk** [vercel/next.js#50337] [Help]
### Summary

The documentation says that turbopack works with @svgr/webpack, but doesn't really want to work.

next.config.js

```
const nextConfig = {
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/i,
      use: ["@svgr/webpack"],
    });

    return config;
  },

  experimental: {
    appDir: true,
    turbo: {
      loaders: {
        ".svg": ["@svgr/webpack"],
      },
    },
  },
};

module.exports = nextConfig;

```

If you run the project without turbopack, everything works fine.

Error run turbopack:

```
Processing image failed
  Failed to parse svg source code for image dimensions
  
  Caused by:
  - Source code does not contain a <svg> root element
```

**Comment by tntchn** [comment]
There is an issue #48140 in turbopack working with @svgr/webpack, so probable need to wait for resolving this

**Comment by ifedayoprince** [comment]
Hello, is this fixed? I'm still running into the error

**Reply by tntchn** [reply]
https://github.com/vercel/turbo/issues/4832#issuecomment-1751407444 Does this work for you?

**Comment by cpotey** [comment]
###   EDIT
  This is sorted now. I was being thick...
  
  I already had an instance of `experimental` in my next.config.js, and I had to put the above turbo code inside it, as I accidentally had two `experimental`s in there - worked after that! Hope this might help someone else look at their code twice!
  
  ---

Also running into the `Error: Unsupported Server Component type: {...}` error still

We've got a shared UI library, and are importing a Footer component, which has SVGs inside it. This Footer component is used in multiple Next.js apps - and it works if I disable turbopack, but errors if enabled.

```js
import Facebook from '@assets/images/icons/social/facebook.svg';
import LinkedIn from '@assets/images/icons/social/linkedin.svg';
import Twitter from '@assets/images/icons/social/twitter.svg';

...

const CommonFooter = ({ children }: FooterProps) => {
  const footerData = {
    ...
    social: [
      {
        id: 1,
        title: 'Twitter',
        img: <Twitter />,
      },
      {
        id: 2,
        title: 'Facebook',
        img: <Facebook />,
      },
      {
        id: 3,
        title: 'Linkedin',
        img: <LinkedIn />,
      },
    ],
  };

  return (
    <footer>
        {children}
        
      <Container>
        <nav>
          {footerData.social.map((link) => (
            <a
              href={link.url}
              target="_blank"
              rel="noreferrer"
              aria-label={link.title}
              key={link.id}
            >
              {link.img}
            </a>
          ))}
        </nav>
      </Container>
    </footer>
  );
};
```

I've got all the usual stuff in my Next.config.js too - including the 

```js 
    turbopack: {
      rules: {
        '*.svg': {
          loaders: ['@svgr/webpack'],
          as: '*.js',
        },
      },
    },
  ```

**Comment by bloodykheeng** [comment]
@penteleichuk 

try this typescript 
next 15
next.config.ts

```
import type { NextConfig } from "next";

const nextConfig: NextConfig = {
  /* config options here */
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      use: [
        {
          loader: "@svgr/webpack",
          options: {
            icon: true
          }
        }
      ]
    });
    return config;
  },
  experimental: {
    turbo: {
      rules: {
        "*.svg": {
          loaders: ["@svgr/webpack"],
          as: "*.js"
        }
      }
    }
  }
};

export default nextConfig;


```


npm i @svgr/webpack

**Reply by seateam** [reply]
非常感谢你，成功了！
Thank you very much, it worked!

**Reply by bloodykheeng** [reply]
Your welcome 😉

**Reply by its-mohammadhosein** [reply]
Worked well, Thank you for being helpful

**Reply by bloodykheeng** [reply]
Your welcome 🤗

**Reply by Manhha3010** [reply]
Bố cứu đời con

**Reply by abdul-irfan-k** [reply]
Thank you, it worked

**Reply by d-zhukov-snotor** [reply]
Thanks!

**Reply by bradleyplaydon** [reply]
Damn, thank you!

**Reply by AhmedReda958** [reply]
Thanks Bro

**Comment by sectsect** [comment]
It might be helpful for others in the future, so just leaving this example here.  
It seems that passing options to `@svgr/webpack` is supported in this format.

### Example for **Next.js v15.3.0**


`15.3.0`: `experimental.turbo` is changed to `turbopack`.



```ts
const nextConfig: NextConfig = {
  turbopack: {
    rules: {
      '*.svg': {
        loaders: [
          {
            loader: '@svgr/webpack',
            options: {
              svgoConfig: {
                plugins: [
                  {
                    name: 'preset-default',
                    params: {
                      overrides: {
                        // customize default plugin options
                        removeViewBox: false,
                      },
                    },
                  },
                  'removeDimensions',
                ],
              },
            },
          },
        ],
        as: '*.js',
      },
    },
  },
}

**Reply by its-mohammadhosein** [reply]
I used to do remove width and height manually

**Reply by vishalx07** [reply]
It only works in development. When I run the build, it throws an error—but if I remove the SVG, everything works fine and the application builds successfully.

![image](https://github.com/user-attachments/assets/e04a827c-dd7a-4eae-875e-a2169fe4de57)

**Reply by sectsect** [reply]
If you're using `next build` (without the `--turbopack` flag), you also need to include the traditional webpack configuration in `next.config.ts`.

**Reply by d-zhukov-snotor** [reply]
thanks, it works ideal
